ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandStore.java

```diff
@@ -112,19 +132,29 @@ public CommandId distributeStatement(
           e
       );
     }
-    return commandId;
+    return status;
   }
 
   /**
    * Poll for new commands, blocking until at least one is available.
    *
    * @return The commands that have been polled from the command topic
    */
-  public ConsumerRecords<CommandId, Command> getNewCommands() {
-    return commandConsumer.poll(Long.MAX_VALUE);
+  public List<QueuedCommand> getNewCommands() {
+    final List<QueuedCommand> queuedCommands = new ArrayList<>();
+    commandConsumer.poll(Duration.ofMillis(Long.MAX_VALUE)).forEach(
+        c -> queuedCommands.add(
+            new QueuedCommand(
+                c.key(),
+                c.value(),
+                commandStatusMap.remove(c.key())
+            )
+        )
+    );
+    return queuedCommands;
   }
 
-  RestoreCommands getRestoreCommands() {
+  public RestoreCommands getRestoreCommands() {
```

Comment:
Why `public`?

Reply:
It's part of the ReplayableQueue interface